Docker 1.11.0 crash with invalid memory address or nil pointer dereference on libnetwork.(*resolver).forwardQueryStart

--------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------

**Output of `docker version`:**

```
Client:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   4dc5990
 Built:        Wed Apr 13 18:38:59 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   4dc5990
 Built:        Wed Apr 13 18:38:59 2016
 OS/Arch:      linux/amd64
```


**Output of `docker info`:**

```
Containers: 8
 Running: 2
 Paused: 0
 Stopped: 6
Images: 426
Server Version: 1.11.0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 568
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 4.2.0-35-generic
Operating System: Ubuntu 15.10
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.704 GiB
Name: ubuntu
ID: 2U4Z:HNRO:CIHL:JXJQ:CMMB:QN2D:WL5O:KK72:UGDQ:RRYQ:W2DH:N72B
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
```

I don't know how to reproduce it, but it already occured 2 times today (I upgraded to 1.11.0 today). Both time Docker crashed with the following error (in syslog):
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x30 pc=0x87d73d]
goroutine 177134 [running]:
github.com/docker/libnetwork.(*resolver).forwardQueryStart(0xc8215d2600, 0x0, 0x0, 0xc820c6ae10, 0x0)
/usr/src/docker/vendor/src/github.com/docker/libnetwork/resolver.go:442 +0x5d
github.com/docker/libnetwork.(*resolver).ServeDNS(0xc8215d2600, 0x0, 0x0, 0xc820c6ae10)
/usr/src/docker/vendor/src/github.com/docker/libnetwork/resolver.go:391 +0xde4
github.com/miekg/dns.(*Server).serve(0xc820b0f450, 0x7f9f18445530, 0xc820abc870, 0x7f9f1cab02f8, 0xc8215d2600, 0xc820f23a00, 0x39, 0x200, 0xc8215b05a8, 0xc82020dd00, ...)
/usr/src/docker/vendor/src/github.com/miekg/dns/server.go:535 +0x7c1
created by github.com/miekg/dns.(*Server).serveUDP
/usr/src/docker/vendor/src/github.com/miekg/dns/server.go:489 +0x3d5
```

--- a/hack/vendor.sh
+++ b/hack/vendor.sh
@@ -30,7 +30,7 @@ clone git github.com/RackSec/srslog 259aed10dfa74ea2961eddd1d9847619f6e98837
 clone git github.com/imdario/mergo 0.2.1
 
 #get libnetwork packages
-clone git github.com/docker/libnetwork v0.7.0-rc.6
+clone git github.com/docker/libnetwork v0.7.0-rc.7
 clone git github.com/armon/go-metrics eb0af217e5e9747e41dd5303755356b62d28e3ec
 clone git github.com/hashicorp/go-msgpack 71c2886f5a673a35f909803f38ece5810165097b
 clone git github.com/hashicorp/memberlist 9a1e242e454d2443df330bdd51a436d5a9058fc4

